packages/framework-core/src/services/read-model-store.ts

```diff
@@ -39,38 +40,61 @@ export class ReadModelStore {
         const readModelName = projectionMetadata.class.name
         const entityInstance = createInstance(entityMetadata.class, entitySnapshotEnvelope.value)
         const readModelID = this.joinKeyForProjection(entityInstance, projectionMetadata)
+        const sequenceKey = this.sequenceKeyForProjection(entityInstance, projectionMetadata)
         this.logger.debug(
           '[ReadModelStore#project] Projecting entity snapshot ',
           entitySnapshotEnvelope,
-          ` to build new state of read model ${readModelName} with ID ${readModelID}`
+          ` to build new state of read model ${readModelName} with ID ${readModelID}`,
+          sequenceKey ? ` sequencing by ${sequenceKey.name} with value ${sequenceKey.value}` : ''
         )
 
         return await retryIfError(
           this.logger,
-          () => this.applyProjectionToReadModel(readModelName, readModelID, projectionMetadata, entityInstance),
+          () =>
+            this.applyProjectionToReadModel(
+              entityInstance,
+              projectionMetadata,
+              readModelName,
+              readModelID,
+              sequenceKey
+            ),
           OptimisticConcurrencyUnexpectedVersionError
         )
       })
     )
   }
 
-  private joinKeyForProjection(entitySnapshot: EntityInterface, projectionMetadata: ProjectionMetadata): UUID {
-    const joinKey = (entitySnapshot as any)[projectionMetadata.joinKey]
+  private joinKeyForProjection(entity: EntityInterface, projectionMetadata: ProjectionMetadata): UUID {
+    const joinKey = (entity as any)[projectionMetadata.joinKey]
     if (!joinKey) {
       throw new InvalidParameterError(
-        `Couldn't find the joinKey named ${projectionMetadata.joinKey} in entity snapshot: ${entitySnapshot}`
+        `Couldn't find the joinKey named ${projectionMetadata.joinKey} in entity snapshot: ${entity}`
       )
     }
     return joinKey
   }
 
+  private sequenceKeyForProjection(
+    entity: EntityInterface,
+    projectionMetadata: ProjectionMetadata
+  ): SequenceKey | undefined {
+    const sequenceKeyName = this.config.readModelSequenceKeys[projectionMetadata.class.name]
+    const sequenceKeyValue = (entity as any)[sequenceKeyName]
+    if (sequenceKeyName && sequenceKeyValue) {
+      return { name: sequenceKeyName, value: sequenceKeyValue }
+    } else {
+      return undefined
+    }
```

Comment:
No need for this `else`
```suggestion
    }
```